### PR TITLE
Enable PSReadline to work on NanoServer

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
PSReadline uses two native Win32 apis to determine font and unicode information which is not available on NanoServer.  Fix is to assume the input is one byte in character in size and not using a truetype font.  This should work for most use cases.

Manually tested this change with our published dockerfile for NanoServer.

Fix https://github.com/PowerShell/PowerShell/issues/3463

### CODE REVIEW

File: src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs
Comment (lzybkr) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs]: We should find a better way - this will cause an exception on every character typed, not good for performance and extremely annoying when debugging.

There is some one-time initialization code - so one less ugly option is to add a P/Invoke call there just to detect if the api is present or not, setting a flag that is checked here.

File: src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs
Comment (lzybkr) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: We should do the same here - detect once so we don't have frequent exceptions.

**Review by lzybkr — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs]: Will make the change

File: src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: Will change

File: src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs]: I looked into putting the check in Readline.Initialize() but since it was a different class I would have to expose two additional members.  Seemed simpler to keep the state within the classes needing those apis.

**Comment by SteveL-MSFT** [comment]
@lzybkr fixed

File: src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs
Comment (lzybkr) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs]: After a closer look, I don't think this is the right fix for this case - custom key bindings need a valid results, you'll just return `\0`.

I think we'll need the Unix variant of this code as a fallback, or potentially use it unconditionally, though that might break some more obscure key bindings.

Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleKeyChordConverter.cs]: There's a comment in the code that GetCharFromConsoleKey() is needed to support chords like Ctrl+[.  Seems like the best approach here is to defer to the Unix code if this API isn't available.  I'll make that change.

**Comment by SteveL-MSFT** [comment]
Manually verified, you can use something like this:

```
docker run -it --entrypoint cmd -v C:\users\slee\repos\PowerShell\src\powershell-win-core\bin\Debug\netcoreapp2.0\win10-x64\publish:c:\powershell microsoft/powershell:nanoserver
```

After the container starts in cmd, you can start powershell using `c:\powershell\powershell`

**Comment by SteveL-MSFT** [comment]
@lzybkr good to go?

File: src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs
Comment (lzybkr) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: If this api is not available, based on the comment, I would expect to see some garbage in the output of `Get-PSReadlineKeyHandler -Bound`, e.g. on this line:
```
Ctrl+]                GotoBrace               Go to matching brace
```
(or if you are using Emacs mode):
```
Ctrl+]           CharacterSearch         Read a character and move the cursor to the next occurence of that character
Ctrl+Alt+]       CharacterSearchBackward Read a character and move the cursor to the previous occurence of that char...
```

Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: I can check next week when back in office where I have Docker and a Nano container, but not sure what you are proposing here when the api is not available.

Comment (lzybkr) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: I'm proposing not degrading the default user experience if possible - which might require a few special cases. I would guess a more general fix isn't possible because this api is that general fix.

Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: Seems like this is a decision the Nano team should make.  If they want the full experience, they need those apis on Nano.  I'll certainly let them know of the limitation.

Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: On a Nano container that line says:
```
Ctrl+Oem6             GotoBrace               Go to matching brace
```

Comment (lzybkr) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: Indeed, not very helpful :)

Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: If you want, I can add specialized code so that if those apis aren't available, we don't have those bindings by default.  Seems like over optimizing for what is today an edge case.  Alternatively, I could have PSReadline fail to load if those apis aren't available even though most of the PSReadline capability seems to run fine on Nano.  Seems like the best option is to document this and have Nano team make the call to add that api.

Comment (lzybkr) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: A few special cases is not over optimizing and is a practical assuming those apis aren't really generally useful.
The default output of `Get-PSReadlineKeyBindings` should not contain `Oem`.

Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs]: I created https://github.com/PowerShell/PowerShell/issues/3890 to track this issue, so we can at least unblock Nano for now.

**Review by lzybkr — APPROVED** [review]

**Pull request merged**
